modules/incentives/src/migration.rs

```diff
@@ -19,29 +19,91 @@
+
+/// Clear all PendingMultiRewards for specific Pool
+pub struct ClearPendingMultiRewards<T, GetPoolId>(PhantomData<T>, PhantomData<GetPoolId>);
+impl<T: Config, GetPoolId: Get<PoolId>> OnRuntimeUpgrade for ClearPendingMultiRewards<T, GetPoolId> {
 	fn on_runtime_upgrade() -> Weight {
+		let pool_id = GetPoolId::get();
 		log::info!(
 			target: "incentives",
-			"ClearDexSavingRewardRates::on_runtime_upgrade execute, will clear Storage DexSavingRewardRates",
+			"ClearPendingMultiRewards::on_runtime_upgrade execute, will clear Storage PendingMultiRewards for Pool {:?}",
+			pool_id,
 		);
 
-		// clear storage DexSavingRewardRates,
-		let _ = DexSavingRewardRates::<T>::clear(u32::max_value(), None);
+		// clear all PendingMultiRewards for specific pool
+		let _ = PendingMultiRewards::<T>::clear_prefix(pool_id, u32::max_value(), None);
 
 		0
 	}
 
 	#[cfg(feature = "try-runtime")]
 	fn post_upgrade() -> Result<(), &'static str> {
-		assert_eq!(DexSavingRewardRates::<T>::iter().count(), 0);
+		let pool_id = GetPoolId::get();
+		assert_eq!(PendingMultiRewards::<T>::iter_prefix(pool_id).count(), 0);
 
 		log::info!(
 			target: "incentives",
-			"ClearDexSavingRewardRates done!",
+			"ClearPendingMultiRewards for Pool {:?} done!",
+			pool_id,
+		);
+
+		Ok(())
+	}
+}
+
+/// Reset rewards record for storage rewards.SharesAndWithdrawnRewards and rewards.PoolInfos at
+/// specific PoolId
+pub struct ResetRewardsRecord<T, GetPoolId>(PhantomData<T>, PhantomData<GetPoolId>);
+impl<T: Config, GetPoolId: Get<PoolId>> OnRuntimeUpgrade for ResetRewardsRecord<T, GetPoolId> {
+	fn on_runtime_upgrade() -> Weight {
+		let pool_id = GetPoolId::get();
+		log::info!(
+			target: "rewards",
+			"ResetRewardsRecord::on_runtime_upgrade execute, will reset Storage SharesAndWithdrawnRewards and PoolInfos for Pool {:?}",
+			pool_id
+		);
+
+		let mut total_share: Balance = Default::default();
+
+		// reset SharesAndWithdrawnRewards
+		for (who, (share, _)) in orml_rewards::SharesAndWithdrawnRewards::<T>::iter_prefix(&pool_id) {
+			orml_rewards::SharesAndWithdrawnRewards::<T>::mutate(&pool_id, &who, |(_, withdrawn_rewards)| {
+				*withdrawn_rewards = WithdrawnRewards::new();
+			});
+
+			total_share = total_share.saturating_add(share);
+		}
+
+		// reset PoolInfos
+		let pool_info = orml_rewards::PoolInfo::<Balance, Balance, CurrencyId> {
+			total_shares: total_share,
+			..Default::default()
+		};
+		orml_rewards::PoolInfos::<T>::insert(&pool_id, pool_info);
+
+		0
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade() -> Result<(), &'static str> {
+		let pool_id = GetPoolId::get();
+		let mut total_share = Balance::default();
+
+		for (_, (share, withdrawn_rewards)) in orml_rewards::SharesAndWithdrawnRewards::<T>::iter_prefix(&pool_id) {
+			assert_eq!(withdrawn_rewards, WithdrawnRewards::new());
+			total_share = total_share.saturating_add(share);
+		}
+
+		assert_eq!(orml_rewards::PoolInfos::<T>::get(&pool_id).total_shares, total_share);
```

Comment:
I think we also need `assert_eq` to check `rewards` of each `PoolInfo` is empty btree map.